src/Components/ProfilePageContextProvider.js

```diff
@@ -17,7 +17,7 @@ export default function ProfilePageContextProvider({ userId, children }) {
     profile: userOwnsProfile ? localUser : profile,
     profileUserId: userId,
     isOwnProfile: userOwnsProfile,
-    isLoading: loading || profileLoading || !localUser?.uid,
```

Comment:
I see how this is required to make profiles and lists visible for users without auth, but it does have the side-effect of making the profile page show in a state that looks like there is no data during cold load, between the time the Firebase user is obtained and the LocalUser is populated.  It is usually super short, so not a huge deal, but not as clean.  I have an idea to fix, though.

In general, a shift where we can't assume there is a localUser on all pages that aren't login or landing is definitely a new thing for our site.  I think a lot of decisions on the 'authenticated' pages may have been made in a world where LocalUser is not empty (for example, what happens when the user likes an anime in a Watchlist they've arrived at via a shared link and they don't have a uid?).  I really love the idea that new people can click a shared link and see content immediately without onboarding, and I think it is absolutely essential to grow the site, so I like this change.  But it might be worth thinking about how we want to handle these un-authenticated users.  I like how they get kicked to the login page if they try to explore around, but another option is that they get authed as a guest _automatically_, and can start using the site right away (skipping onboarding, basically).  Then when they click like or create a watchlist, it actually works!  And they still get the upsells to create a permanent account, which can covert them into full users.  This is good as-is for now, but I think this might be worth thinking about.

Reply:
Thanks for catching this...I meant to confirm with you that this change didn't have any weird side-effects. Whoops! 😬

And yeah, I agree - that sounds like the best way to go about it.  My only hesitation is we'll blow through loads of guest accounts...but I'm pretty sure Firebase has a way to delete unused/old users, so we can address that issue at a later date.